generator/main.cpp

```diff
@@ -132,36 +137,205 @@ int main(int argc, char *argv[])
+        }
+    }
+
+    if (!qtHeaders.isEmpty()) {
+        /* Look for the 'standard' directories, favouring the first found
+         * in the qt-headers list.
+         *
+         * NOTE: use of a list here is intended for the project build workflows
+         * which apparently have to assemble the headers from difference
+         * installation directories.  This could be avoided by using
+         * --include-paths instead!
+         *
+         */
+        QStringList dirList;
+        QStringList requiredModules(coreModules);
+
+        foreach (const QString &dir, qtHeaders.split(path_splitter))
+            if (QDir(dir).exists())  {
+                QStringList remaining(requiredModules);
+
+                foreach (const QString &mod, requiredModules) {
+                    const QString modpath(dir + "/Qt" + mod);
+                    if (QDir(modpath).exists()) {
+                        includes << modpath;
+                        remaining.removeAll(mod);
+                    }
+                }
+
+                requiredModules = remaining;
+                dirList << dir;
+            } else
+                qWarning((dir + " Qt header directory not found")
+                                            .toLocal8Bit().constData());
+
+        /* Add the top level directories at the end; this simulates the legacy
+         * behaviour.
+         */
+        if (!minimal)
+            includes += dirList;
+
+        /* Warn if standard directories are not found: */
+        if (!requiredModules.isEmpty()) {
+            QString sep("");
+            QString errMsg("WARNING: missing core Qt modules:");
+
+            foreach (const QString &mod, requiredModules) {
+                errMsg += sep + " Qt" + mod;
+                sep = ",";
+            }
+
+            if (args.contains("core-error"))
+                qFatal(errMsg.toLocal8Bit().constData());
+            else
+                qWarning(errMsg.toLocal8Bit().constData());
+        }
+    } else if (includes.isEmpty()) {
+        /* This is never true unless --minimal is given because "." is always
+         * added to includes first.
+         */
+        qWarning("No directories to scan: use --include-paths or --qt-headers");
+        displayHelp(&*gs);
+    }
+
+    std::cout << "-------------------------------------------------------------" << std::endl;
+    std::cout << "Scanning Qt headers from (in this order):" << std::endl;
+    foreach (const QString &dir, includes) {
+        std::cout << "  " << dir.toUtf8().constData();
+        if (!QDir(dir).exists())
+            std::cout << " [DIRECTORY DOES NOT EXIST]";
+        std::cout << std::endl;
+    }
+    std::cout << "-------------------------------------------------------------" << std::endl;
+
     printf("Please wait while source files are being generated...\n");
 
     printf("Parsing typesystem file [%s]\n", qPrintable(typesystemFileName));
+    fflush(stdout);
```

Comment:
Note that `std::endl` also does a flush, though I guess `printf` was used here throughout the code, and you don't want to replace it all...